executor/ddl.go

```diff
@@ -200,10 +203,65 @@ func (e *DDLExec) executeAlterDatabase(s *ast.AlterDatabaseStmt) error {
 }
 
 func (e *DDLExec) executeCreateTable(s *ast.CreateTableStmt) error {
+	if s.TemporaryKeyword == ast.TemporaryLocal {
+		return e.createSessionTemporaryTable(s)
+	}
+
 	err := domain.GetDomain(e.ctx).DDL().CreateTable(e.ctx, s)
 	return err
 }
 
+func (e *DDLExec) createSessionTemporaryTable(s *ast.CreateTableStmt) error {
```

Comment:
Is it better to modify the method `ddl.CreateTableWithInfo` , it can reduce many dup codes like alloc table id, generate table info, etc...

Reply:
Local temporary table is not a normal table, it doesn't need to do the DDL schema change and store table info.

Reply:
I checked the function ddl.CreateTable, ddl.CreateTableWithInfo, most codes can be reused here. We can just return if local temporary if founded before creating table in tikv. Current modification is easy to lost some checks. For example,
create temporary table if not exists will still return error because you didn't check IfNotExists in statement

Reply:
Check the table options will be done in some other PR.
That will add up to a lot of work, and (more importantly) the tests should also be included.